swarm/api/api_test.go

```diff
@@ -433,3 +433,52 @@ func TestDecryptOrigin(t *testing.T) {
 		}
 	}
 }
+
+func TestDetectContentType(t *testing.T) {
+	// internally use http.DetectContentType, so here are test cases only about fallback to file extension check
+	testDetectContentType(t, "file.css", "body {background-color: orange}", "text/css; charset=utf-8")
+	testDetectContentType(t, "file-empty.css", "", "text/css; charset=utf-8")
+	testDetectContentType(t, "file.pdf", "", "application/pdf")
+	testDetectContentType(t, "file.md", "", "text/plain; charset=utf-8")
+	testDetectContentType(t, "file-with-unknown-content.strangeext", "", "text/plain; charset=utf-8")
+	testDetectContentType(t, "file-with-text.strangeext", "Lorem Ipsum", "text/plain; charset=utf-8")
+	testDetectContentType(t, "file-no-extension", "Lorem Ipsum", "text/plain; charset=utf-8")
+	testDetectContentType(t, "file-no-extension-no-content", "", "text/plain; charset=utf-8")
+	testDetectContentType(t, "css_with_html_inside.css", "<!doctype html><html><head></head><body></body></html>", "text/css; charset=utf-8")
+}
+
+func testDetectContentType(t *testing.T, fileName, content, expectedContentType string) {
+	f, err := tempFileWithContent(fileName, content)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(f.Name())
```

Comment:
You could avoid the need for a temp file by modifying `DetectContentType` to take `(filename string, data io.ReadSeeker)` and just use a `bytes.Reader` in the tests.

Reply:
indeed, it will simplify everything